src/workloads/query/PercentilesWindow.yml

```diff
@@ -0,0 +1,173 @@
+    Type: CrudActor
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Repeat: 10
+          Database: *db
+          Collection: *coll
+          Operations:
+          - OperationName: aggregate
+            OperationCommand:
+              Pipeline: [{
+                $setWindowFields: {
+                  sortBy: {date: 1},
+                  output: {
+                    p: {
+                      $minN: {n: 1000, input: "$data" },
+                      window: { ^Parameter: {Name: "Documents", Default: {documents: [-500, 500]}}
+                      }}
+                  }}
+              }]
+
+Actors:
+- Name: CreateDataset
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *maxPhases
+      PhaseConfig:
+        Threads: 1
+        Repeat: 1
+        Database: *db
+        CollectionCount: 1
+        DocumentCount: 100000
+        BatchSize: 1000
+        Document:
+          _id: {^Inc: {start: 0}}
+          # This is the field we will be computing percentiles on.
+          data: {^RandomDouble: {distribution: normal, mean: 50.0, sigma: 10.0}}
+          # We will use this field to sort our documents and create range windows.
+          date: {^RandomDate: {min: "2013-01-01", max: "2023-01-01"}}
+
+- ActorFromTemplate:
+    TemplateName: ComputePercentile
+    TemplateParameters:
+      Name: "percentile_small_window"
+      OnlyActiveInPhase: 1
+
+- ActorFromTemplate:
+    TemplateName: ComputePercentile
+    TemplateParameters:
+      Name: "percentile_large_window"
+      Documents: { documents: [0, "unbounded"]}
+      OnlyActiveInPhase: 2
+
+- ActorFromTemplate:
+    TemplateName: ComputePercentile
+    TemplateParameters:
+      Name: "percentile_cumulative_t_digest"
+      Documents: { documents: ["unbounded", 0]}
+      OnlyActiveInPhase: 3
+
+# The next phases compute percentiles over range windows. We do not expect to compare the
+# runtime of range and document based windows, since the sizes of these windows can vary
+# greatly. However, we still want to cover all use cases of window functions, and did compare
+# $percentile runtime to $minN for all types of windows during development.
+- ActorFromTemplate:
+    TemplateName: ComputePercentile
+    TemplateParameters:
+      Name: "percentile_small_range_window"
+      # If the unit was larger than 'day' we would have nearly all the documents in every window,
+      # which would not test having a smaller bounded window.
+      Documents: { range: [-500, 500], unit: "day"}
+      OnlyActiveInPhase: 4
+
+- ActorFromTemplate:
+    TemplateName: ComputePercentile
+    TemplateParameters:
+      Name: "percentile_large_range_window"
+      # We want to test larger windows, so we increase the unit to 'month'.
```

Comment:
the comment is somewhat confusing to me: we test larger windows by the virtue of them being unbounded, no? if I understand it correctly, the unit choice would affect the pattern of how many values are added/removed before recomputing the percentile (which is a great test to have, but could you please confirm that my understanding is correct here?)

Reply:
yes the unit choice does affect the pattern of how many values are added/removed, but it is also required for the range windows that have a time range. So to test different patterns of adding/removing values we can have 2 tests. One we run a [0, "unbounded"] window with the 'year' unit to pull in as many document as possible and the other we can run on a tighter unit like 'day'.